ext/spl/tests/ArrayObject/appending_object.phpt

```diff
@@ -0,0 +1,28 @@
+--TEST--
+Cannot append to ArrayObject if backing value is an object
+--EXTENSIONS--
+spl
+--FILE--
+<?php
+class MyClass {}
+$c = new MyClass();
+
+$ao = new ArrayObject($c);
+
+try {
+    $ao->append('no-key');
+    var_dump($c);
+} catch (\Throwable $e) {
+    echo $e::class, ': ', $e->getMessage(), PHP_EOL;
+}
+
+try {
+    $ao->append('no-key');
+    var_dump($c);
+} catch (\Throwable $e) {
+    echo $e::class, ': ', $e->getMessage(), PHP_EOL;
+}
```

Comment:
Did you meant to use `$ao[] = 'no-key'` in one of these?

Reply:
Yes indeed.